mrtt-ui/src/components/CausesOfDeclineForm.js

```diff
@@ -0,0 +1,389 @@
+  RadioGroup,
+  TextField,
+  Typography
+} from '@mui/material'
+import { useForm, useFieldArray, Controller } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+
+import {
+  Form,
+  FormQuestionDiv,
+  MainFormDiv,
+  NestedFormSectionDiv,
+  SectionFormTitle,
+  SubTitle,
+  SubTitle2
+} from '../styles/forms'
+import { ErrorText } from '../styles/typography'
+import ButtonSubmit from './ButtonSubmit'
+import { causesOfDecline } from '../data/questions'
+import { causesOfDeclineOptions } from '../data/causesOfDeclineOptions'
+import { mapDataForApi } from '../library/mapDataForApi'
+
+function CausesOfDeclineForm() {
+  const validationSchema = yup.object().shape({
+    lossKnown: yup.boolean(),
+    causesOfDecline: yup
+      .array()
+      .of(
+        yup.object().shape({
+          mainCauseLabel: yup.string(),
+          mainCauseAnswers: yup.array().of(
+            yup.object().shape({
+              mainCauseAnswer: yup.string(),
+              levelOfDegredation: yup.string().required()
+            })
+          ),
+          subCauses: yup.array().of(
+            yup.object().shape({
+              subCauseLabel: yup.string(),
+              subCauseAnswers: yup.array().of(
+                yup.object().shape({
+                  subCauseAnswer: yup.string(),
+                  levelOfDegredation: yup.string().required()
+                })
+              )
+            })
+          )
+        })
+      )
+      .min(1)
+      .required('Select at least one cause of decline')
+      .default([])
+  })
+
+  const formOptions = { resolver: yupResolver(validationSchema) }
+
+  // get functions to build form with useForm() hook
+  const { control, formState, watch, handleSubmit } = useForm(formOptions)
+  const { errors } = formState
+  const {
+    fields: causesOfDeclineFields,
+    append: causesOfDeclineAppend,
+    remove: causesOfDeclineRemove,
+    update: causesOfDeclineUpdate
+  } = useFieldArray({ name: 'causesOfDecline', control })
+  const lossKnownWatcher = watch('lossKnown')
+
+  const [isSubmitting, setisSubmitting] = useState(false)
+  const [isError, setIsError] = useState(false)
+  const { siteId } = useParams()
+  const url = `${process.env.REACT_APP_API_URL}/sites/${siteId}/registration_answers`
+
+  // big function with many different cases for Q4.2 due to the nesting involved in this question type
+  const handleCausesOfDeclineOnChange = ({
+    event,
+    mainCauseLabel,
+    subCauseLabel,
+    childOption,
+    secondaryChildOption
+  }) => {
+    const mainCauseIndex = causesOfDeclineFields.findIndex(
+      (cause) => cause.mainCauseLabel === mainCauseLabel
+    )
+    const currentMainCause = causesOfDeclineFields[mainCauseIndex]
+    const subCauseIndex = currentMainCause?.subCauses?.findIndex(
+      (subCause) => subCause.subCauseLabel === subCauseLabel
+    )
+    const currentSubCause = currentMainCause?.subCauses?.[subCauseIndex]
+
```

Comment:
nice naming everything

Reply:
K, gunna stop actual code reviewing to hit that deadline. Just focusing on syncing now.